base/utf16.jl

```diff
@@ -128,7 +530,7 @@ convert(T::Type{UTF16String}, data::AbstractArray{Int16}) =
 
 function convert(T::Type{UTF16String}, bytes::AbstractArray{UInt8})
     isempty(bytes) && return UTF16String(UInt16[0])
-    isodd(length(bytes)) && throw(ArgumentError("odd number of bytes"))
+    isodd(length(bytes)) && throw(UnicodeError("odd number of bytes"))
```

Comment:
What do you suggest then?  Change the UnicodeError into an ArgumentError for now?